docs/reference/api/connectivity/drivers/EMACMemoryManager.md

```diff
@@ -0,0 +1,24 @@
+## EMAC memory manager
+
+The Ethernet MAC memory manager class provides abstracted memory interface toward memory modules used in different network stacks. The EMAC device driver uses the memory interface to handle memory related operations, such as allocation, free and data manipulation. The [EMAC](emac.html) drivers use this memory manager to reserve space for network packets.
+
+Memory buffer chains store the data on the memory interface. The data passed in either direction either may be contiguous (a single-buffer chain) or may consist of multiple buffers. You can chain the buffers using a singly linked list.
+
+### Usage
+
+The EMAC driver can allocate memory either from a memory pool or from the heap (contiguous memory). By preference, link input memory should be allocated from the pool, but if contiguous memory is required by the hardware, it can be allocated from the heap.
```

Comment:
Query: Is this saying that you, the user, should allocate link input memory from the pool unless the hardware requires contiguous memory, in which case you should allocate the link input memory from the heap?

Reply:
If the `user` is the driver, then yes.

This is an API that is used by the driver, not Mbed OS user.